lib/inertia_rails/renderer.rb

```diff
@@ -91,16 +85,15 @@ def merge_props(shared_props, props)
 
     def computed_props
       # rubocop:disable Style/MultilineBlockChain
-      merge_props(shared_data, props)
-        .then do |merged_props| # Always keep errors in the props
+      @props
+        .tap do |merged_props| # Always keep errors in the props
```

Comment:
other part of the fix (just making sure we aren't unnecessarily calling merge_props twice)